plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -5065,6 +5079,29 @@ private Answer execute(MigrateVolumeCommand cmd) {
                         volumePath = vmMo.getVmdkFileBaseName(disk);
                     }
             }
+            if (cmd.getNewIops() != null) {
+                String vmwareDocumentation = "https://kb.vmware.com/s/article/68164";
+                Long newIops = cmd.getNewIops();
+                VirtualDisk disk = vmMo.getDiskDevice(volumePath, true).first();
```

Comment:
This method will try to get the file name based on the volumePath, to do this, it will try to strip the extension from the path. However, the volumePath comes from the database already stripped of the extension. 
This is generally not an issue, but if the file name has a (`.`) in it, it will get split and the method will fail.  We should either refactor that method to take this into account, or create a new one for this use case.

Reply:
It's not the prettiest code I've ever written, but it's done.